src/seq.rs

```diff
@@ -601,23 +654,41 @@ impl fmt::Debug for Event {
     }
 }
 
-/// Low level methods to set/get data on an Event. Don't use these directly, use generic methods on Event instead.
-pub trait EventData {
+/// Low level method to check if event type supports a given data type. Don't use directly, use generic methods on Event instead.
+pub trait HasEventData {
```

Comment:
I'm not really happy about the split into three traits and the associated type on SetEventData. If we instead had an extra `Event::new_borrowed` function for the particular case of borrowing `&[u8]`, would that help?